modules/storage-filesystem/snapshot.go

```diff
@@ -42,54 +42,71 @@ func (m *StorageFileSystemModule) StoreSnapshot(ctx context.Context, snapshot *s
-	if err := ctx.Err(); err != nil {
-		return errors.Wrapf(err, "restore snapshot aborted, invalid context")
-	}
-
-	metaPath := m.makeMetaFilePath(snapshotID)
-
-	metaData, err := os.ReadFile(metaPath)
+func (m *StorageFileSystemModule) RestoreSnapshot(ctx context.Context, className, snapshotId string) error {
+	snapshot, err := m.loadSnapShotMeta(ctx, className, snapshotId)
 	if err != nil {
-		return errors.Wrapf(err, "Could not read snapshot meta file %v", metaPath)
-	}
-	var snapshot snapshots.Snapshot
-	if err := json.Unmarshal(metaData, &snapshot); err != nil {
-		return errors.Wrapf(err, "Could not unmarshal snapshot meta file %v", metaPath)
+		return errors.Wrapf(err, "load snapshot meta")
 	}
 
 	for _, srcRelPath := range snapshot.Files {
 		if err := ctx.Err(); err != nil {
 			return errors.Wrapf(err, "restore snapshot aborted, system might be in an invalid state")
 		}
-		if err := m.copyFile(m.dataPath, m.makeSnapshotDirPath(snapshotID), srcRelPath); err != nil {
+		if err := m.copyFile(m.dataPath, m.makeSnapshotDirPath(className, snapshotId), srcRelPath); err != nil {
 			return errors.Wrapf(err, "restore snapshot aborted, system might be in an invalid state: file %v", srcRelPath)
 		}
 	}
 	return nil
 }
 
-func (m *StorageFileSystemModule) SetMetaStatus(ctx context.Context, className, snapshotID, status string) error {
-	// TODO implement
-	return nil
+func (m *StorageFileSystemModule) loadSnapShotMeta(ctx context.Context, className, snapshotID string) (*snapshots.Snapshot, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, errors.Wrapf(err, "restore snapshot aborted, invalid context")
+	}
+
+	metaPath := m.makeMetaFilePath(className, snapshotID)
+
+	metaData, err := os.ReadFile(metaPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not read snapshot meta file %v", metaPath)
+	}
+	var snapshot snapshots.Snapshot
+	if err := json.Unmarshal(metaData, &snapshot); err != nil {
+		return nil, errors.Wrapf(err, "Could not unmarshal snapshot meta file %v", metaPath)
+	}
+
+	return &snapshot, nil
 }
 
 func (m *StorageFileSystemModule) GetMetaStatus(ctx context.Context, className, snapshotID string) (string, error) {
-	// TODO implement
-	return "", fmt.Errorf("file does not exist")
+	snapshot, err := m.loadSnapShotMeta(ctx, className, snapshotID)
+	if err != nil {
+		return "", errors.Wrapf(err, "load snapshot meta")
+	}
+
+	return string(snapshot.Status), nil
 }
 
-func (m *StorageFileSystemModule) DestinationPath(className, snapshotID string) string {
-	// TODO implement
-	return ""
+func (m *StorageFileSystemModule) SetMetaStatus(ctx context.Context, className, snapshotID, status string) error {
+	snapshot, err := m.loadSnapShotMeta(ctx, className, snapshotID)
+	if err != nil {
+		return errors.Wrapf(err, "load snapshot meta")
+	}
+
+	snapshot.Status = snapshots.Status(status)
+
+	if err := m.saveMeta(snapshot); err != nil {
+		return err
```

Comment:
here we don't need to wrap error as we do this in line 100?

Reply:
I think 100 is the load error, and 105 is the save error?  I updated the message to be more clear